pydfy/template/src/partials/kpi.html

```diff
@@ -0,0 +1,10 @@
+<div class="pf-kpi border-solid border border-gray-100">
+    <div class="pf-kpi-title bg-gray-200 text-lg font-bold text-center align-top">{{ component.title }}</div>
+    <div class="pf-number-content mt-2 text-center">
+        <div class="text-3xl">{{ "{:,}".format(component.current) }}</div>
+        <div class="text-xs mb-2">{{ "Goal: {:,} ({:+.2f} %)".format(component.goal,
+                                                                     (component.current -
+                                                                      component.goal) /
+                                                                      component.goal * 100) }}</div>
```

Comment:
The layout looks good! While experimenting I did stumble upon a zero-division error when the target equals 0. Since this becomes tedious to code in these html templates, my suggestion would be to move this logic to a `@property` on the `KPI` called `_percent` for instance. Also see the `_base64_data` field on the `Image` component. Then you can access e.g. the `_percent` column and return "Infinity" when the target equals 0, which is how PowerBI handles it.

(Another feature you could consider adding for this use case is: a flag called `use_value` with default `False`. If `True`, we display the absolute difference instead of relative. This does require changing the template to handle this as well though.)

Reply:
Thanks, I appreciate your time and suggestions. Maybe the suggested feature here could be a new issue?

Reply:
No worries, I'll add a new issue for the suggestion.